lib/RegisterController.js

```diff
@@ -71,3 +78,29 @@ export class RegisterController {
     }
   }
 }
+
+async function _verifyTurnstileToken({token}) {
+  // dummy secret key, load actual key from config
+  const SECRET_KEY = '1x0000000000000000000000000000000AA';
+  // TODO: load url from config
+  const url = 'https://challenges.cloudflare.com/turnstile/v0/siteverify';
+  const payload = {
+    secret: SECRET_KEY,
+    response: token,
+    // how to get remoteIP?
+    remoteip: ''
+  };
+  try {
+    const result = await httpClient.post(url, payload);
+    if(result.data.success) {
+      // captcha was successful, do nothing
+      return;
+    } else {
+      // what error to throw?
+      throw new Error('Turnstile token not validated.');
+    }
+  } catch(e) {
+    // what do we do if there is an error with the turnstile call itself?
+    console.dir(e);
```

Comment:
We likely need to assume the visitor is not a bot...and be sure the form still works if/when Turnstile fails. That should be easy to test in the network console by blocking the turnstile URLs.